File: lp2/bomerp-backend/src/main/java/pe/edu/upeu/bomerp/acopio/parametros/service/MineroService.java
package pe.edu.upeu.bomerp.acopio.parametros.service;

import pe.edu.upeu.bomerp.acopio.parametros.dto.MineroRequest;
import pe.edu.upeu.bomerp.acopio.parametros.dto.MineroResponse;
import java.util.List;

public interface MineroService {
    List<MineroResponse> listar();
    MineroResponse obtener(Long idMinero);
    MineroResponse crear(MineroRequest request);
    MineroResponse actualizar(Long idMinero, MineroRequest request);
    void eliminar(Long idMinero);
}

File: lp2/sitra-oro-frontend/src/app/features/acopio/mineros/minero.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiService } from '../../../core/api/api.service';
import { Minero, MineroRequest } from './minero';

@Injectable({ providedIn: 'root' })
export class MineroService {
  private readonly http = inject(HttpClient);
  private readonly url = inject(ApiService).buildUrl('/api/v1/acopio/mineros');
  listar(): Observable<Minero[]> { return this.http.get<Minero[]>(this.url); }
  obtener(id: number): Observable<Minero> { return this.http.get<Minero>(`${this.url}/${id}`); }
  crear(request: MineroRequest): Observable<Minero> { return this.http.post<Minero>(this.url, request); }
  actualizar(id: number, request: MineroRequest): Observable<Minero> { return this.http.put<Minero>(`${this.url}/${id}`, request); }
  eliminar(id: number): Observable<void> { return this.http.delete<void>(`${this.url}/${id}`); }
}
